src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -37,57 +39,404 @@ const ActorInit En_Cow_InitVars = {
+
+            this->actionFunc = EnCow_Idle;
+
+            if (!(gSaveContext.weekEventReg[22] & 1) && (CURRENT_DAY != 1) &&
+                (EN_COW_TYPE(thisx) == EN_COW_TYPE_ABDUCTED)) {
+                Actor_MarkForDeath(&this->actor);
+                return;
+            }
+
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_COW, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, this->actor.shape.rot.y, 0,
+                               EN_COW_TYPE_TAIL);
+
+            this->animationTimer = Rand_ZeroFloat(1000.0f) + 40.0f;
+            this->animationCycle = 0;
+            this->actor.targetMode = 6;
+
+            D_801BDAA4 = 0;
+            func_801A5080(4);
+            break;
+        case EN_COW_TYPE_TAIL:
+            SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06004C30, NULL, this->jointTable, this->morphTable,
+                               COW_LIMB_MAX);
+            Animation_PlayLoop(&this->skelAnime, &D_06004348);
+
+            this->actor.update = EnCow_UpdateTail;
+            this->actor.draw = EnCow_DrawTail;
+            this->actionFunc = EnCow_DoTail;
+
+            EnCow_SetTailPos(this);
+
+            this->actor.flags &= ~EN_COW_FLAG_IS_TAIL;
+            this->animationTimer = Rand_ZeroFloat(1000.0f) + 40.0f;
+            break;
+    }
+
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->flags = 0;
+
+    gSaveContext.weekEventReg[87] &= (u8)~1;
+}
+
+void EnCow_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnCow* this = THIS;
+
+    if (this->actor.params == EN_COW_TYPE_DEFAULT) { //! @bug EN_COW_TYPE_ABDUCTED do not destroy their cylinders
+        Collider_DestroyCylinder(globalCtx, &this->colliders[0]);
+        Collider_DestroyCylinder(globalCtx, &this->colliders[1]);
+    }
+}
+
+void EnCow_UpdateAnimation(EnCow* this, GlobalContext* globalCtx) {
+    if (this->animationTimer > 0) {
+        this->animationTimer--;
+    } else {
+        this->animationTimer = Rand_ZeroFloat(500.0f) + 40.0f;
+        Animation_Change(&this->skelAnime, &D_060001CC, 1.0f, this->skelAnime.curFrame,
+                         Animation_GetLastFrame(&D_060001CC), ANIMMODE_ONCE, 1.0f);
+    }
+    if (this->actor.xzDistToPlayer < 150.0f) {
+        if (!(this->flags & EN_COW_FLAG_PLAYER_HAS_APPROACHED)) {
+            this->flags |= EN_COW_FLAG_PLAYER_HAS_APPROACHED;
+            if (this->skelAnime.animation == &D_060001CC) {
+                this->animationTimer = 0;
+            }
+        }
+    }
+    this->animationCycle++;
+    if (this->animationCycle > 0x30) {
+        this->animationCycle = 0;
+    }
+
+    if (this->animationCycle < 0x20) {
+        this->actor.scale.x = ((Math_SinS(this->animationCycle * 0x400) * (1.0f / 100.0f)) + 1.0f) * 0.01f;
+    } else {
+        this->actor.scale.x = 0.01f;
+    }
+
+    if (this->animationCycle >= 0x11) {
```

Comment:
Sorry, I meant my comment to apply to this one too.

Reply:
Oh oops, I should've caught that